dbfit-java/core/src/main/java/dbfit/api/DbQuery.java

```diff
@@ -0,0 +1,32 @@
+package dbfit.api;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+/**
+ * An object representing an executable database operation.
+ * Executing the operation returns a ResultSet result.
+ *
+ */
+public interface DbQuery extends AutoCloseable {
```

Comment:
> What are the benefits of having separate interfaces vs a more complex API?

It's a general principle, some description and examples here: https://springframework.guru/principles-of-object-oriented-design/interface-segregation-principle. But it's a good question - are we on the right balance with these interfaces?